src/libraries/SwapMath.sol

```diff
@@ -70,7 +70,7 @@ library SwapMath {
                     // `amountIn` is capped by the target price
                     sqrtPriceNextX96 = sqrtPriceTargetX96;
                     feeAmount = _feePips == MAX_SWAP_FEE
-                        ? amountIn
+                        ? 0 // 0 as amountRemainingLessFee == 0
```

Comment:
I think in the issue they said to keep it as amountIn and just comment that it will always be 0. But I see that it doesn't cost extra gas for us anyways
But also maybe make the comment more clear like 
// Always 0 because if amountRemainingLessFee >= amountIn with the max fee, amountIn must be 0 (making both amountRemainingLessFee and feeAmount 0).